src/utils/self_test/self_test.c

```diff
@@ -42,32 +60,176 @@ struct st_master_endpt {
 	int32_t test_completed;
 };
 
-static const char * const crt_st_msg_type_str[] = { "EMPTY",
-						    "IOV",
-						    "BULK_PUT",
-						    "BULK_GET" };
+const struct {
+	char identifier;
+	char *short_name;
+	char *long_name;
+	enum crt_st_msg_type type;
+} transfer_type_map[] = {{'e', "E", "EMPTY", CRT_SELF_TEST_MSG_TYPE_EMPTY},
+			 {'i', "I", "IOV", CRT_SELF_TEST_MSG_TYPE_IOV},
+			 {'b', "Bp", "BULK_PUT",
+			  CRT_SELF_TEST_MSG_TYPE_BULK_PUT},
+			 {'r', "Bg", "BULK_GET",
+			  CRT_SELF_TEST_MSG_TYPE_BULK_GET} };
+
+#define MSG_TYPE_STR(str, id)					\
+	{							\
+	int _i;							\
+	for (_i = 0; _i < 4; _i++) {				\
+		if (id == transfer_type_map[_i].type) {		\
+			str =  transfer_type_map[_i].long_name;	\
+			break;					\
+		}						\
+	}	\
+	}
+
+#define	TST_HIGH	0x01	/* test in higher direction */
+#define	TST_LOW		0x02	/* test in lower direction */
+#define	TST_OUTPUT	0x10	/* print results */
+struct status_feature {
+	char	*name;
+	int	 value;
+	float	 scale;
+	int	 flag;
+	char	*description;
+};
 
-/* User input maximum values */
-#define SELF_TEST_MAX_REPETITIONS (0x40000000)
-#define SELF_TEST_MAX_INFLIGHT (0x40000000)
-#define SELF_TEST_MAX_LIST_STR_LEN (1 << 16)
-#define SELF_TEST_MAX_NUM_ENDPOINTS (UINT32_MAX)
+static struct status_feature status[] = {
+	{"bw", 0, 0, TST_HIGH | TST_LOW | TST_OUTPUT, "Bandwidth"},
+	{"tp", 0, 0, TST_HIGH | TST_LOW | TST_OUTPUT, "Throughput"},
+	{"av", 0, 0, TST_HIGH | TST_LOW | TST_OUTPUT, "Averages"},
+	{"sd", 0, 0, TST_HIGH | TST_LOW | TST_OUTPUT, "Standard Deviations"},
+	{"min", 0, 0, TST_LOW | TST_OUTPUT, "Minimum"},
+	{"med25", 0, 0, TST_HIGH | TST_LOW | TST_OUTPUT, "Medium 25"},
+	{"med50", 0, 0, TST_HIGH | TST_LOW | TST_OUTPUT, "Medium 50"},
+	{"med75", 0, 0, TST_HIGH | TST_LOW | TST_OUTPUT, "Medium 75"},
+	{"max", 0, 0, TST_HIGH | TST_OUTPUT, "Maximum"},
+};
 
 /* Global shutdown flag, used to terminate the progress thread */
-static int g_shutdown_flag;
+struct global_params {
```

Comment:
_Moving global variables into a single structure_ -- is this to make it easier to print, debug, pass the globals to other functions?

Reply:
Alex ask me to put globals into a single structure.  This should make future debugging easier, just having to dump out a single structure and having everything shown in a consistent fashion.